rdr_service/tools/tool_libs/consent_validation_report.py

```diff
@@ -131,171 +134,152 @@
+                  AND DATE(ps.{authored_field}) <= "{end_date}"
+                  AND cf.sync_status = 1
+    """
+
+# Filter for generating stats on file issues resolved by retransmission (OBSOLETE consent_file entries)
+# The last modified timestamp for an OBSOLETE record should reflect when it was moved into OBSOLETE status; include
+# resolutions up to the specified end date for this report
+ALL_RESOLVED_SQL = """
+           SELECT cf.participant_id,
+                  cf.type,
+                  DATE(cf.modified) AS resolved_date
+           FROM consent_file cf
+           WHERE cf.sync_status = 3 AND DATE(cf.modified) <= "{end_date}"
+"""
 
 # TODO:  Remove this when we expand consent validation to include CE consents
-VIBRENT_SQL_FILTER = ' AND ps.participant_origin = "vibrent"'
-
-# Define the allowable --report-type arguments and their associated SQL.
-REPORT_TYPES = {
-    # Daily uploads = validation for all consents authored on the report date + missing files flagged on the report date
-    'daily_uploads':  CONSENT_REPORT_SQL_BODY + DAILY_NEW_CONSENTS_SQL_FILTER + VIBRENT_SQL_FILTER,
-    # Unresolved errors = Any consent_file entries still in a NEEDS_CORRECTING state (all-time)
-    'unresolved_errors':  CONSENT_REPORT_SQL_BODY + ALL_UNRESOLVED_ERRORS_SQL_FILTER + VIBRENT_SQL_FILTER
-}
+VIBRENT_SQL_FILTER = ' AND ps.participant_origin = "vibrent" '
+
+# Weekly report SQL for validation burndown counts
+CONSENTED_PARTICIPANTS_COUNT_SQL = """
+    SELECT COUNT(DISTINCT ps.participant_id)  consented_participants
+    FROM participant_summary ps
+    JOIN participant p ON p.participant_id = ps.participant_id
+         AND p.is_test_participant = 0 and (p.is_ghost_id is null or not p.is_ghost_id) and p.hpo_id != 21
+    WHERE DATE(ps.consent_for_study_enrollment_first_yes_authored) <= "{end_date}"
+"""
+
+VALIDATED_PARTICIPANTS_COUNT_SQL = """
+    SELECT COUNT(DISTINCT cf.participant_id) validated_participants
+    FROM consent_file cf
+    JOIN  participant ps ON ps.participant_id = cf.participant_id
+         AND ps.is_test_participant = 0 and (ps.is_ghost_id is null or not ps.is_ghost_id) and ps.hpo_id != 21
+    WHERE DATE(cf.created) <= "{end_date}"
+"""
+
+# Define the allowable --report-type arguments
+REPORT_TYPES = ['daily_uploads', 'weekly_status']
+
+
+class SafeDict(dict):
+    """
+    See: https://stackoverflow.com/questions/17215400/format-string-unused-named-arguments
+    Used with str.format_map() to allow partial formatting/replacement of placeholder values in a string
+    without incurring KeyError. E.g.: '{lastname}, {firstname} {lastname}'.format_map(SafeDict(lastname='Bond'))
+    yields the partially formatted result: 'Bond, {firstname} Bond'
+
+     This helper class will be used when formatting SQL template strings that need to have some of their placeholders
+     filled in before the values for the others can be determined.
+    """
+    def __missing__(self, key):
+        return '{' + key + '}'
```

Comment:
Can we use an f-string here instead of concatenation and anywhere else where we use it or `format()` method?

Reply:
I can change this here (grabbed this from stackoverflow).  For me, using `format()` is sometimes a compromise when the length of the f-string is getting problematic.  But I'll look for cases where that wasn't a factor.